src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonTypeInfo.cs

```diff
@@ -404,22 +405,38 @@ internal void VerifyMutable()
 
         private protected volatile bool _isConfigured;
         private readonly object _configureLock = new object();
+        private ExceptionDispatchInfo? _cachedConfigureError;
 
         internal bool IsConfigured => _isConfigured;
 
         internal void EnsureConfigured()
         {
-            if (_isConfigured)
-                return;
+            if (!_isConfigured)
+                ConfigureLocked();
 
-            lock (_configureLock)
+            void ConfigureLocked()
             {
-                if (_isConfigured)
-                    return;
+                _cachedConfigureError?.Throw();
```

Comment:
Moving this check inside the `ConfigureLocked` method to ensure the outer `EnsureConfigured` method can be inlined in the hot path.